### DISCUSSION TITLE
Is a controller required?

### DISCUSSION THREAD

**Discussion by phealy** [cvonk/OPNpool#8] [Q&A] [answered]
This looks like an awesome project, and I'm hoping to build it so I don't have to buy a controller :) thanks for building it and open sourcing it!

I have a few questions I was hoping you might have thoughts on:

1. I currently have a Pentair IntelliFlow VSF, MasterTemp 250 heater, and an iChlor 30 - all of which support RS-485. Hardware-wise it looks to me like I should be able to add this on w/o the SunTouch or equivalent (I can get the 15V power/ground from the iChlor) - do you think that will work, or am I likely to run into software issues where the controller is assumed to be present?

1. I also have a dumb SR Smith plx60 light controller [(PDF manual link)](https://www.srsmith.com/media/178643/plx_pls_manual_06_2018.pdf) (dumb in the sense that the RF remote is just on/off and you rotate color modes by powering on/off in sequence) that I'd like to control. At this point I think (as I'm not very practiced with electronics of this type) I'd be best served by just adding on a separate smart relay to Home Assistant instead of trying to attach to the OPNpool, as it doesn't look like there are any super easy pinouts to add that already built on the board. Did I miss anything?

1. I saw in the readme that you have some extra partly assembled units if you're in the Silicon Valley area. If you're willing to ship one elsewhere in the US, I'd be happy to buy one to save some time (covering S&H as well, of course).

Thanks again!

**Comment by cvonk** [comment]
1. Yes, OPNpool uses the controller.  This was a deliberate design choice.  The key advantage is that the pool system will be functional with or without OPNpool.  So, if you ever want to sell the house, you're not stuck with a custom system controlling the pool. Having said that, of cause it is totally doable to build your own controller. It will just be a very different project.  If you poke around the web, you will find various attempts.

2. The controller has several relays.  I use one to control my underwater light.

3. Sorry I don't ship.

/c

**Comment by evercape** [comment]
Thanks - this project looks great. 

Why is the controller needed? If one ignores the relays, shouldn't your RS485 integration via MQTT be able to at least read if not write all the information?

Thanks
M

**Comment by cvonk** [comment, marked as answer]
> Why us the controller needed
To quote myself: "This was a deliberate design choice. The key advantage is that the pool system will be functional with or without OPNpool. So, if you ever want to sell the house, you're not stuck with a custom system controlling the pool. Having said that, of cause it is totally doable to build your own controller. It will just be a very different project. If you poke around the web, you will find various attempts."

**Reply by evercape** [reply]
Thanks Coert, fair enough and I did read your quote in the documentation.

I run quite a number of pumps and chlorinators, and have literally no direct pool automation, other than a KNX bus system with integration to HA.

I have installed temperature and pH sensors, as well as actuators that control the power supply to pumps and chlorinators. Those KNX actuators have an integrated Amp/Watt sensor, so based on that I can "second guess" if a pump is priming, or running in normal speed, and if running too low then disconnect chlorinators. 

All of the pumps have a display and on-pump control, so there is a way to control things in an old fashion way. I could in theory open up the Pentair Intelliflow and try to add 4 channel relays to trigger the pre-programmed modes 1-4. What I am most interested in is RPM read and write. Scheduling I can do from HA by simply cutting power, if needed, or alternatively triggering the start/stop command.

My preferred choice would be an ESPHome chip with RS484 to Wifi, and everything within HA/ESP.

I looked around at [nodes-poolController](https://github.com/tagyoureit/nodejs-poolController), however, that requires separate hardware. I have a few pumps, not all in one location, so not ideal even with my main system running on Proxmox with VMs. 

[CenturyVSPump](https://github.com/gazoodle/CenturyVSPump) seems to maybe work, but haven't figured it out. Also, I am based in Africa, so access to hardware not as easy as in the US West Coast. 

There is an [yoctupuce](https://www.yoctopuce.com/EN/article/controlling-an-intelliflo-pump-from-home-assistant) Integration but again probably overkill with MQTT and also extra multiple hardware. 

Lastly, I am old school database programmer. Never done python, learned a bit of yaml. Touched C++ last more than 35 years ago....that gives you an idea ;-)

I will see if I can learn from your integration and documentation, which is very thorough. Well done, and thanks!
